### PR TITLE
Update docs to include information about the strategies that use quickfix

### PR DESCRIPTION
**Pull request by codeinabox**
Arises from #14

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -117,6 +117,18 @@ disable this:
 let g:test#preserve_screen = 1
 ```
 
+### Quickfix Strategies
```
Comment (janko-m) [inline README.md]: Empty line missing between the title and the first paragraph.

File: README.md [hunk 2]
```diff
@@ -117,6 +117,18 @@ disable this:
...
+If you want your test results to appear in the quickfix window, use one of the 
+following strategies:
+
+ * Make
+ * Neomake
+ * MakeGreen
+ * Dispatch.vim
+
+When Dispatch.vim is installed, it will be used to determine the compiler and 
+errorformat settings for the test runner.
```
Comment (janko-m) [inline README.md]: I would prefer for this paragraph to be something like:

> Even if you're using a quickfix strategy other than Dispatch.vim, it's still recommended you have Dispatch.vim installed, as the strategies will automatically use it to find and apply the correct Vim compiler, ensuring the test output is correctly parsed for the quickfix window.

There is no need to mention `errorformat` when we already mention "compiler", as setting `errorformat` is the only thing the compiler does (other than setting `makeprg` which we don't use).

**Comment by codeinabox** [comment]
Have updated based on feedback. Do we need to explain to users that not all compilers are supported out of the box by Dispatch?

**Comment by codeinabox** [comment]
@janko-m any further feedback on this?

**Comment by janko-m** [comment]
It would be great to explain what you mentioned, and that compilers come from the Vim distribution and plugins, not from Dispatch.vim.

It would also be good to document the g:dispatch_compilers variable, as it's really useful when the test command prefix doesn't match compiler's makeprg. But if you're not sure what that variable does exactly (it took myself some time to understand), I can fill in that part.

**Comment by codeinabox** [comment]
Using #166 as reference I have added an example of g:dispatch_compilers. Feel free to tweak anything I've written.

**Comment by codeinabox** [comment]
@janko-m have ready through it I think is thorough enough but let me know if I have missed anything and feel free to alter

**Comment by codeinabox** [comment]
@janko-m are you happy for me to merge this?

**Pull request merged**